source/userguide/text_preparation_pipeline.rst

```diff
@@ -0,0 +1,665 @@
+| SeparateAposS                                       | Adds a space before 's.                                                                          | "mindmeld's code"                     | "mindmeld 's code"            |
++-----------------------------------------------------+--------------------------------------------------------------------------------------------------+---------------------------------------+-------------------------------+
+| ReplacePunctuationAtWordStartWithSpace              | Replaces special characters infront of words with a space.                                       | "HI %#++=-=SPERO"                     | "HI SPERO"                    |
++-----------------------------------------------------+--------------------------------------------------------------------------------------------------+---------------------------------------+-------------------------------+
+| ReplacePunctuationAtWordEndWithSpace                | Replaces special characters following words with a space.                                        | "How%+=* are++- you^^%"               | "How are you"                 |
++-----------------------------------------------------+--------------------------------------------------------------------------------------------------+---------------------------------------+-------------------------------+
+| ReplaceSpecialCharsBetweenLettersAndDigitsWithSpace | Replaces special characters between letters and digits with a space.                             | "Coding^^!#%24 hours#%7 days"         | "Coding 24 hours 7 days"      |
++-----------------------------------------------------+--------------------------------------------------------------------------------------------------+---------------------------------------+-------------------------------+
+| ReplaceSpecialCharsBetweenDigitsAndLettersWithSpace | Replaces special characters between digits and letters with a space.                             | "Coding 24^^!#%%hours 7##%days"       | "Coding 24 hours 7 days"      |
++-----------------------------------------------------+--------------------------------------------------------------------------------------------------+---------------------------------------+-------------------------------+
+| ReplaceSpecialCharsBetweenLettersWithSpace          | Replaces special characters between letters and letters with a space.                            | "Coding all^^!#%%hours seven##%days"  | "Coding all hours seven days" |
++-----------------------------------------------------+--------------------------------------------------------------------------------------------------+---------------------------------------+-------------------------------+
+
+The last 5 rules above remove special characters in different contexts. These special characters can be specified in the config using the key, :attr:`keep_special_chars`.
```

Comment:
I think it would be better to name the 5 here since if we add more to the table, we have to also update it here.

Reply:
Sure, added it in.